libs/shared/src/services/native/app-update.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable } from '@angular/core';
+import { AppUpdate, AppUpdateAvailability, AppUpdateInfo } from '@capawesome/capacitor-app-update';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AppUpdateService {
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  constructor() {}
+
+  /** Checks for app updates and starts a flexible update if allowed */
+  async checkForUpdates() {
+    try {
+      const appUpdateInfo: AppUpdateInfo = await AppUpdate.getAppUpdateInfo();
+      if (
+        appUpdateInfo.updateAvailability === AppUpdateAvailability.UPDATE_AVAILABLE &&
+        appUpdateInfo.flexibleUpdateAllowed
+      ) {
+        await this.startFlexibleUpdate();
+      }
+    } catch (error) {
+      console.error('Error checking for updates:', error);
+    }
+  }
+
+  /** Starts a flexible update process */
+  private async startFlexibleUpdate() {
```

Comment:
All looks good to me, happy to use the flexible update pathway. I think you can probably remove the console logs as capacitor produces its own logs during AppUpdate operations when running on native device